### PR TITLE
Replaces ashy spellblade with new item: bloody knuckles

### PR DESCRIPTION
**Pull request by LazennG**
# Document the changes in your pull request
vampiric_nuckles
lazenn makes another stupid blood themed item. Bloody knuckles are a melee item that mark targets after impact. Marked targets can be hit again to root them in place; people for 1 second and everything else for 6. Marking targets this way puts it on a cooldown for 10 seconds if the last marked target was a human and 6 seconds otherwise. Additionally, the knuckles grant 2 abilities with a cooldown of 15 seconds. One ability marks people within 7 tiles of sight if they're standing on blood and the other roots marked targets within 8 spaces.

Came up with this idea after thinking about what could help miners in lavaland and what would still be somewhat thematic with the boss whose drops are all combat-oriented.

spellblade was always a shit drop and ashy spellblade was lame coping

# Wiki Documentation

at https://wiki.yogstation.net/wiki/Megafauna#Loot_3
replace the line about the spellblade with something like
bloody knuckles that can root victims in place and do 50 damage to nonhumanoids.

# Changelog

:cl:  @Chubbygummibear, @ToasterBiome, @cuackles 
rscadd: Added new item: bloody knuckles 
rscdel: Removes ashy spellblade from bg loot pool
imageadd: sprites for knuckles, the mark, and root effect 
/:cl:
thanks chubbygummibear and baiomu for coding everything functional and cuackles for sprites

### CODE REVIEW

**Comment by SomeguyManperson** [comment]
image

File: code/datums/action.dm [hunk 1]
```diff
@@ -531,6 +531,18 @@
 		owner.playsound_local(box, 'sound/misc/box_deploy.ogg', 50, TRUE)
 		QDEL_NULL(box)
 
+/datum/action/item_action/visegrip
+	name = "Vise Grip"
```
Comment (SomeguyManperson) [inline code/datums/action.dm:535]:
```suggestion
	name = "Vice Grip"
```

**Comment by LazennG** [comment]
> image

i have 0 clue how to get airshoe commits out of the fucking list

File: code/datums/action.dm (hunk 1 above)
Comment (LazennG) [reply code/datums/action.dm:535]: image
bri'ish?

**Comment by Marmio64** [comment]
spellblade was a shit drop until you wanted to validhunt/do antag shit and insta delimb people when RNG blesses you (rip zombies)

File: code/datums/status_effects/knuckleroot.dm [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		duration = 60
+		owner.adjustBruteLoss(50)
+	return ..()
+
+/datum/status_effect/roots/proc/owner_moved()
+	return COMPONENT_MOVABLE_BLOCK_PRE_MOVE
+
+/datum/status_effect/roots/on_remove()
+	if(!owner.stat)
+		to_chat(owner, "Your bindings are gone!")
```
Comment (tattax) [inline code/datums/status_effects/knuckleroot.dm]:
```suggestion
		to_chat(owner, span_notice("Your bindings are gone!"))
```

i like my colored chat texts

Comment (tattax) [reply code/datums/status_effects/knuckleroot.dm]: could add some italics tbh

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 3]
```diff
@@ -1112,6 +1112,70 @@ GLOBAL_LIST_EMPTY(bloodmen_list)
 	log_combat(user, L, "took out a blood contract on", src)
 	qdel(src)
 
+/obj/item/melee/knuckles
```
Comment (tattax) [inline code/modules/mining/lavaland/necropolis_chests.dm:1121]:
```suggestion
#define COOLDOWN 150
#define COOLDOWN_HUMAN 100
#define COOLDOWN_ANIMAL 60
/obj/item/melee/knuckles
```

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 4]
```diff
@@ -1112,6 +1112,70 @@ GLOBAL_LIST_EMPTY(bloodmen_list)
...
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "bloodyknuckle"
+	lefthand_file = 'icons/mob/inhands/weapons/melee_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
+	item_state = "knuckles"
+	w_class = WEIGHT_CLASS_SMALL
+	force = 18
+	var/cooldown = 150
+	var/cdh = 100
+	var/cda = 60
```
Comment (tattax) [inline code/modules/mining/lavaland/necropolis_chests.dm]:
```suggestion
```

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 5]
```diff
@@ -1112,6 +1112,70 @@ GLOBAL_LIST_EMPTY(bloodmen_list)
...
+		var/valid_reaching = FALSE
+		for(var/mob/living/L in view(7, U))
+			for(var/obj/effect/decal/cleanable/B in range(0,L))
+				if(istype(B, /obj/effect/decal/cleanable/blood )|| istype(B, /obj/effect/decal/cleanable/trail_holder))
+					valid_reaching = TRUE
+					L.apply_status_effect(STATUS_EFFECT_KNUCKLED)
+		if(!valid_reaching)
+			to_chat(U, span_warning("There's nobody to use this on!"))
+			return
+		next_reach = world.time + cooldown
```
Comment (tattax) [inline code/modules/mining/lavaland/necropolis_chests.dm]:
```suggestion
		next_reach = world.time + COOLDOWN
```

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 6]
```diff
@@ -1112,6 +1112,70 @@ GLOBAL_LIST_EMPTY(bloodmen_list)
...
+			return
+		var/valid_casting = FALSE
+		for(var/mob/living/L in view(8, U))
+			if(L.has_status_effect(STATUS_EFFECT_KNUCKLED))
+				valid_casting = TRUE
+				L.apply_status_effect(/datum/status_effect/roots)
+		if(!valid_casting)
+			to_chat(U, span_warning("There's nobody to use this on!"))
+			return
+		next_grip = world.time + cooldown
```
Comment (tattax) [inline code/modules/mining/lavaland/necropolis_chests.dm]:
```suggestion
		next_grip = world.time + COOLDOWN
		#undef COOLDOWN
		#undef COOLDOWN_HUMAN
		#undef COOLDOWN_ANIMAL
```

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 7]
```diff
@@ -1112,6 +1112,70 @@ GLOBAL_LIST_EMPTY(bloodmen_list)
...
+			return
+		if(next_knuckle > world.time)
+			to_chat(user, span_warning("The knuckles aren't ready to mark yet."))
+			return
+		else
+			L.apply_status_effect(STATUS_EFFECT_KNUCKLED)
+			if(ishuman(L))
+				next_knuckle = world.time + cdh
+			if(!ishuman(L))
+				next_knuckle = world.time + cda
```
Comment (tattax) [inline code/modules/mining/lavaland/necropolis_chests.dm]:
```suggestion
			if(ishuman(L))
				next_knuckle = world.time + COOLDOWN_HUMAN
				return
			next_knuckle = world.time + COOLDOWN_ANIMAL
```

File: code/datums/status_effects/knuckleroot.dm [hunk 8]
```diff
@@ -0,0 +1,34 @@
+/datum/status_effect/roots
+	id = "roots"
+	alert_type = /obj/screen/alert/status_effect/roots
+	var/icon/cube
+	duration = 10
```
Comment (ToasterBiome) [inline code/datums/status_effects/knuckleroot.dm]:
```suggestion
	duration = 1 SECONDS
```

File: code/datums/status_effects/debuffs.dm [hunk 9]
```diff
@@ -1064,4 +1064,29 @@
 	else if(isanimal(owner))
 		var/mob/living/simple_animal/S = owner
 		for(var/i in S.damage_coeff)
-			S.damage_coeff[i] /= power
\ No newline at end of file
+			S.damage_coeff[i] /= power
+
+/datum/status_effect/knuckled
+    id = "knuckle_wound"
+    duration = 100
```
Comment (ToasterBiome) [inline code/datums/status_effects/debuffs.dm]:
```suggestion
    duration = 10 SECONDS
```

File: code/datums/status_effects/knuckleroot.dm [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+	RegisterSignal(owner, COMSIG_MOVABLE_PRE_MOVE, .proc/owner_moved)
+	if(!owner.stat)
+		to_chat(owner, span_userdanger("You're held in place by some hellish force! Fight back while you can!"))
+	cube = icon('icons/effects/effects.dmi', "leghold")
+	var/icon/size_check = icon(owner.icon, owner.icon_state)
+	cube.Scale(size_check.Width(), size_check.Height())
+	owner.add_overlay(cube)
+	owner.remove_status_effect(STATUS_EFFECT_KNUCKLED)
+	if(!ishuman(owner))
+		duration = 60
```
Comment (ToasterBiome) [inline code/datums/status_effects/knuckleroot.dm]:
```suggestion
		duration = 6 SECONDS
```

**Review by ToasterBiome — DISMISSED** [review]

**Comment by KillerOrcaCora** [comment]
Doing gods work amen.

**Comment by JamieD1** [comment]
@grimm-hollowknight

**Review by nesisknesis — APPROVED** [review]

**Pull request merged**